cli/tests/activate.bats

```diff
@@ -1164,15 +1141,27 @@ EOF
 
 # bats test_tags=activate,activate:inplace-modifies,activate:inplace-modifies:fish
 @test "'flox activate' modifies the current shell (fish)" {
-  export FLOX_FEATURES_USE_CATALOG=false
-  project_setup
+  project_setup_pkgdb
+
   # set profile scripts
-  sed -i -e "s/^\[profile\]/${HELLO_PROFILE_SCRIPT//$'\n'/\\n}/" "$PROJECT_DIR/.flox/env/manifest.toml"
+  echo "$HELLO_PROFILE_SCRIPT" >>"$PROJECT_DIR/.flox/env/manifest.toml"
   # set a hook
-  sed -i -e "s/^\[hook\]/${VARS_HOOK_SCRIPT//$'\n'/\\n}/" "$PROJECT_DIR/.flox/env/manifest.toml"
+  echo "$VARS_HOOK_SCRIPT" >>"$PROJECT_DIR/.flox/env/manifest.toml"
   # set vars
-  sed -i -e "s/^\[vars\]/${VARS//$'\n'/\\n}/" "$PROJECT_DIR/.flox/env/manifest.toml"
-  "$FLOX_BIN" install hello
+  echo "${VARS}" >>"$PROJECT_DIR/.flox/env/manifest.toml"
+
+  # update the lockfile to avoid re-locking
+  manifest_lock="$(cat "$PROJECT_DIR/.flox/env/manifest.lock")"
+  manifest_lock="$(
+    jq ".manifest.profile = $(tomlq '.profile' "$PROJECT_DIR/.flox/env/manifest.toml")" <(echo "$manifest_lock")
+  )"
+  manifest_lock="$(
+    jq ".manifest.hook = $(tomlq '.hook' "$PROJECT_DIR/.flox/env/manifest.toml")" <(echo "$manifest_lock")
+  )"
+  manifest_lock="$(
+    jq ".manifest.vars = $(tomlq '.vars' "$PROJECT_DIR/.flox/env/manifest.toml")" <(echo "$manifest_lock")
+  )"
+  echo "$manifest_lock" >"$PROJECT_DIR/.flox/env/manifest.lock"
```

Comment:
nit: seems like it would be easier to just commit this `manifest.lock` somewhere

Reply:
pulled out a manifest/lock for this